src/synchronizer.rs

```diff
@@ -29,13 +30,15 @@ use crate::synchronizer::SynchronizerError::*;
 ///   - `H` - hasher used for checksum calculation
 ///   - `N` - serializer scratch space size
 ///   - `SD` - sleep duration in nanoseconds used by writer during lock acquisition (default 1s)
+///   - `WL` - optional write locking to prevent multiple writers. (default [`Disabled`])
 pub struct Synchronizer<
     H: Hasher + Default = WyHash,
     const N: usize = 1024,
     const SD: u64 = 1_000_000_000,
+    WL = Disabled,
```

Comment:
To me it makes more sense to move `WL` right after `H`, so that it's easier to use synchronizer with `N` and `SD` defaults.

Reply:
Updated. Note, this is a breaking change to any code specifying the `N` or `SD` parameters, so we'll need a major version bump when releasing.